### ISSUE TITLE
Bracket Pair Colorization: Consider not coloring angle brackets in JSX

### ISSUE THREAD

**Issue by IllusionMH** [microsoft/vscode#131003]
At this moment colors of `<` and `>` in tags depends on nesting in blocks/parentheses. In my opinion it's more confusing then helpful. That's because all these tags have same color depending on blocks/parentheses but not on how deeply component nested (as each tag has both `<` and `>`)

On the other hand it looks like `<`/`>` in generics are not colored at all.
![image](https://user-images.githubusercontent.com/1312662/129741021-e6f17c4a-fa0c-4f33-80c8-c18e72ca5d6e.png)

I think it would be more useful not to color `<`/`>`  for JSX elements and, maybe, add coloring for generics (to help distinguish nested generics) if possible.

<details>
<summary>Code</summary>

Language: TypeScript React
```tsx
declare const MyComponent: React.FC;
declare const condition: boolean;

type Generic<T extends Array<any>> = T[0];

const A = () => {
    if (condition) {
        if (condition) {
            return <MyComponent />;
        }
        return <MyComponent />;
    }
    return condition
        ? <MyComponent />
        : (
            <MyComponent>
                <p>
                    <MyComponent children={ <MyComponent /> } />;
                </p>
            </MyComponent>
        );
};
```

</details>

**Comment by hediet** [comment]
I agree, but this is specific to TypeScript and its grammar.

I'm quite certain that the bracket pair colorizer extension does the same as our native implementation:
JSX elements are tagged as jsx-element-language, everything else as TSX.
In TSX, `<` and `>` do not form a bracket pair. In jsx-element-language they do.

I guess it does not make sense for JSX/HTML to register `<`...`>` as bracket pair.

**Comment by IllusionMH** [comment]
Hmm... BPC2 extension doesn't apply iterating colors for `<`/`>` in JavaScript/JavaScript React/TypeScript React therefore my request to avoid this (or not to "start coloring" if built-in can be considered as continuation).

<details>
<summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/1312662/129748472-50e10a89-73d2-41af-9cb9-8747bd97cfcc.png)
![image](https://user-images.githubusercontent.com/1312662/129748599-cf8cd9cd-9f8c-4643-b3f7-ee950e14a7be.png)

No BPC2 class on `<` but on `{` it's available.
![image](https://user-images.githubusercontent.com/1312662/129750632-bb09dee1-016e-449b-bd0f-9879dee9eeb8.png)
</details>

**Comment by CoenraadS** [comment]
BPC2 has manual override to ignore `<>` as they caused lots of problems

https://github.com/CoenraadS/Bracket-Pair-Colorizer-2/blob/271162f4a8ef85d940dda85a880e0703eba79a4b/src/textMateLoader.ts#L86-L87

**Comment by hediet** [comment]
This would be fixed by #130996.
